File: FrontEnd/src/app/usuarios/models/usuario.model.ts
import { Role } from './role.models';
export class Usuario {
    id: number;
    username: string;
    password: string;
    confirmarPassword: string;
    nomeCompleto: string;
    email: string;
    enabled: boolean;
    foto: string;
    urlFoto: string;
    roles: Role;
}

export class UsuarioDTO {
    username: string;
    nomeCompleto: string;
    email: string;
    roles: Role;
}

File: BackEnd/src/main/java/br/com/blinkdev/leadsponge/models/usuario/UsuarioModel.java
package br.com.blinkdev.leadsponge.models.usuario;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.hateoas.RepresentationModel;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UsuarioModel extends RepresentationModel<UsuarioModel> {
    private Long id;
    private String username;
    private String nomeCompleto;
    private String email;
    private Boolean enabled;
}
